File: PlankCooking/models/Product.cs

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlankCooking.Models {

 
 
 public partial class Product
    {
        [Key]
        public int ProductID { get; set; }

        [ForeignKey("CategoryID")]
        public Category Category { get; set; }
        public int CategoryID { get; set; }

        [Required, StringLength(100)]
        public string Name { get; set; }
        
        public string Description { get; set; }

        [Required, DataType(DataType.Currency)]
        public decimal Price { get; set; }

        [StringLength(50)]
        public string PriceDescription { get; set; }

   
        public Int16 SortOrder { get; set; }

        [Required]
        public bool Active { get; set; }

        [Required]
        public decimal Ounces { get; set; }

        [StringLength(500)]
        public string ImagePath { get; set; }

        [DataType(DataType.Currency), Required]
        public decimal HandlingCost { get; set; }

        [Required]
        public bool TaxExempt { get; set; }

        public string SKU {get; set;}

    }
}

File: PlankCooking/ClientApp/app/models/Product.ts

export class Product {
   
    productID: number;
    categoryID: number;
    name: string;
    description: string;
    price: number;
    priceDescription: string;
    sortOrder: number;
    active: boolean;
    ounces: number;
    imagePath: string;
    handlingCost: number;
    taxExempt: boolean;
    sKU: string;


    constructor() {

        this.productID = 0;
        this.categoryID = 0;
        this.name = "";
        this.description = "";
        this.price = 0;
        this.priceDescription = "";
        this.sortOrder = 0;
        this.active = false;
        this.ounces = 0;
        this.imagePath = "";
        this.handlingCost = 0;
        this.taxExempt = false;
        this.sKU = "";

    }
}